### PR TITLE
Bug 1852964: account for nil DaemonSet returned from library-go

### PR DESCRIPTION
**Pull request by gabemontero**
See https://github.com/openshift/library-go/blob/master/pkg/operator/resource/resourceapply/apps.go#L125

Just going to account for nil rets from library-go in ocm-o

/assign @adambkaplan

### CODE REVIEW

File: pkg/operator/sync_openshiftcontrollermanager_v311_00.go [hunk 1]
```diff
@@ -95,9 +95,17 @@ func syncOpenShiftControllerManager_v311_00_to_latest(c OpenShiftControllerManag
 
 	// our configmaps and secrets are in order, now it is time to create the DS
 	// TODO check basic preconditions here
+	var progressingMessages []string
 	actualDaemonSet, _, err := manageOpenShiftControllerManagerDeployment_v311_00_to_latest(c.kubeClient.AppsV1(), c.recorder, operatorConfig, c.targetImagePullSpec, operatorConfig.Status.Generations, forceRollout, c.proxyLister)
 	if err != nil {
-		errors = append(errors, fmt.Errorf("%q: %v", "deployment", err))
+		msg := fmt.Sprintf("%q: %v", "deployment", err)
```
Comment (sttts) [inline pkg/operator/sync_openshiftcontrollermanager_v311_00.go:101]: what's the goal here?

File: pkg/operator/sync_openshiftcontrollermanager_v311_00.go [hunk 2]
```diff
@@ -95,9 +95,17 @@ func syncOpenShiftControllerManager_v311_00_to_latest(c OpenShiftControllerManag
...
+		progressingMessages = append(progressingMessages, msg)
+		errors = append(errors, fmt.Errorf(msg))
+	}
+
+	// library-go func called by manageOpenShiftControllerManagerDeployment_v311_00_to_latest can return nil with errors
+	if actualDaemonSet == nil {
+		goto PROCESS_RETURN
```
Comment (sttts) [inline pkg/operator/sync_openshiftcontrollermanager_v311_00.go:108]: https://www.cs.utexas.edu/users/EWD/ewd02xx/EWD215.PDF

File: pkg/operator/sync_openshiftcontrollermanager_v311_00.go (hunk 1 above)
Comment (gabemontero) [reply pkg/operator/sync_openshiftcontrollermanager_v311_00.go:101]: to include error information in the progressing message

File: pkg/operator/sync_openshiftcontrollermanager_v311_00.go (hunk 2 above)
Comment (gabemontero) [reply pkg/operator/sync_openshiftcontrollermanager_v311_00.go:108]: I'll move off of goto

**Comment by gabemontero** [comment]
update to no longer use `goto` pushed @sttts - thanks

File: pkg/operator/sync_openshiftcontrollermanager_v311_00.go [hunk 3]
```diff
@@ -153,6 +160,11 @@ func syncOpenShiftControllerManager_v311_00_to_latest(c OpenShiftControllerManag
 	operatorConfig.Status.ObservedGeneration = operatorConfig.ObjectMeta.Generation
 	resourcemerge.SetDaemonSetGeneration(&operatorConfig.Status.Generations, actualDaemonSet)
 
+	return processReturn(c, errors, originalOperatorConfig, operatorConfig)
+
+}
+
+func processReturn(c OpenShiftControllerManagerOperator, errors []error, originalOperatorConfig, operatorConfig *operatorapiv1.OpenShiftControllerManager) (bool, error) {
```
Comment (adambkaplan) [inline pkg/operator/sync_openshiftcontrollermanager_v311_00.go:167]: nit - perhaps rename to `syncReturn` or `finishSync`

**Comment by adambkaplan** [comment]
/approve

@gabemontero nit on the name of the return function, otherwise looks good.

File: pkg/operator/sync_openshiftcontrollermanager_v311_00.go (hunk 3 above)
Comment (gabemontero) [reply pkg/operator/sync_openshiftcontrollermanager_v311_00.go]: switched to `syncReturn` @adambkaplan

**Comment by gabemontero** [comment]
update of method name pushed @adambkaplan

**Pull request merged**